src/Features/CSharp/Portable/CodeRefactorings/NodeSelectionHelpers.cs

```diff
@@ -18,8 +20,22 @@ internal static async Task<SyntaxNode> GetSelectedDeclarationOrVariableAsync(Cod
             // MemberDeclaration: member that can be declared in type (those are the ones we can pull up) 
             // VariableDeclaratorSyntax: for fields the MemberDeclaration can actually represent multiple declarations, e.g. `int a = 0, b = 1;`.
             // ..Since the user might want to select & pull up only one of them (e.g. `int a = 0, [|b = 1|];` we also look for closest VariableDeclaratorSyntax.
-            return await context.TryGetRelevantNodeAsync<MemberDeclarationSyntax>().ConfigureAwait(false) as SyntaxNode ??
-                await context.TryGetRelevantNodeAsync<VariableDeclaratorSyntax>().ConfigureAwait(false);
+            var memberDeclaration = await context.TryGetRelevantNodeAsync<MemberDeclarationSyntax>().ConfigureAwait(false);
+            if (memberDeclaration == null)
+            {
+                return await context.TryGetRelevantNodeAsync<VariableDeclaratorSyntax>().ConfigureAwait(false);
+            }
+
+            if (memberDeclaration is not FieldDeclarationSyntax fieldDeclaration)
+            {
+                return memberDeclaration;
+            }
+
+            // since a FieldDeclarationSyntax doesn't have an associated Symbol (which we will want later), we want the variable declaration they ask for
+            // find the variable declarator closest to the start of the span
+            var span = context.Span;
+            return fieldDeclaration.Declaration.Variables.OrderBy(
+                v => Math.Min(Math.Abs(v.SpanStart - span.Start), Math.Abs(v.Span.End - span.Start))).First();
```

Comment:
Not this?

```suggestion
                v => Math.Min(Math.Abs(v.SpanStart - span.Start), Math.Abs(v.Span.End - span.End))).First();
```

At least it seems odd that we're not being symmetrical here?

Reply:
I agree, however currently NodeSelectionHelpers must return a single variable declarator, so we need to choose one of the fields to return and act on. Perhaps we could alter the refactoring features that depend on this to accept multiple member nodes, but it seems out of the scope of this PR.

Currently I think choosing the closest field to the start of the selection is a good way to choose since in the case of selections over multiple, the first/leftmost field in the selection scope would be chosen. 

@CyrusNajmabadi @jasonmalinowski I'd be happy to discuss other potential ways to decide in edge cases related to the example above as well.

Reply:
> I agree, however currently NodeSelectionHelpers must return a single variable declarator,

In this case, i would consider this a failure condition.  THere isn't a single correct variable declarator to return, so it shouldn't return anything.

Reply:
> THere isn't a single correct variable declarator to return, so it shouldn't return anything.

What about the following cases:
 `$$public static int Foo, Bar;`
 `public static int Foo, Bar;$$`
In both, the selection doesn't specifically point to a particular field out of the two. Would these also be ambiguous and return nothing since we can't tell? If that is the case then we could avoid sorting by only giving a valid variable declarator if the selection span overlaps exactly one variable declarator span.

Reply:
> If that is the case then we could avoid sorting by only giving a valid variable declarator if the selection span overlaps exactly one variable declarator span.

That seems reasonable to me.